pkg/aws/aws.go

```diff
@@ -286,27 +286,99 @@ func (c *Client) ClusterSharedIamOperation(operation string) error {
 		if err != nil {
 			return errors.Wrap(err, "failed to create shared cluster role")
 		}
-		log.Infof("BDD >> created shared iam role: %s", awssdk.StringValue(role.Arn))
+		log.Infof("BDD >> created shared iam role: %s", aws.StringValue(role.Arn))
 
-		policy, err := PutManagedPolicy(roleName, clusterSharedPolicy, "shared cluster policy", policyDocument, iamClient)
+		policy, err := PutManagedPolicy(roleName, clusterSharedPolicy, "shared cluster policy", policyDocument, c.IAMClient)
 		if err != nil {
 			return errors.Wrap(err, "failed to create shared cluster managed policy")
 		}
-		log.Infof("BDD >> created shared iam policy: %s", awssdk.StringValue(policy.Arn))
+		log.Infof("BDD >> created shared iam policy: %s", aws.StringValue(policy.Arn))
 	case "remove":
-		err := DeleteManagedPolicy(clusterSharedPolicy, iamClient)
+		err := DeleteManagedPolicy(clusterSharedPolicy, c.IAMClient)
 		if err != nil {
 			return errors.Wrap(err, "failed to delete shared cluster role")
 		}
 
-		err = DeleteIAMRole(roleName, iamClient)
+		err = DeleteIAMRole(roleName, c.IAMClient)
 		if err != nil {
 			return errors.Wrap(err, "failed to delete shared cluster managed policy")
 		}
 	}
 	return nil
 }
+
+func (c *Client) GetEksVpc() (string, error) {
+	clusterName, err := util.GetClusterName()
+	if err != nil {
+		return "", err
+	}
+	input := &eks.DescribeClusterInput{
+		Name: aws.String(clusterName),
+	}
+	result, err := c.EKSClient.DescribeCluster(input)
+	if err != nil {
+		return "", err
+	}
+	return aws.StringValue(result.Cluster.ResourcesVpcConfig.VpcId), nil
+}
+
+func (c *Client) DnsNameInHostedZoneID(dnsName string, hostedZoneID string) error {
+	time.Sleep(120 * time.Second)
+
+	recordValue, err := c.GetDNSRecord(dnsName, hostedZoneID)
+	if err != nil {
+		if fmt.Sprintf("more than 1 records for hostedZoneID %s with dnsName %s", hostedZoneID, dnsName) == string(err.Error()) {
+			fmt.Printf("records for hostedZoneID %s with dnsName %s exists", hostedZoneID, dnsName)
+			return nil
+		} else if recordValue != "" {
+			fmt.Printf("records for hostedZoneID %s with dnsName %s exists", hostedZoneID, dnsName)
+			return nil
+		} else {
+			return errors.Errorf("records for hostedZoneID %s with dnsName %s doesn't exists", hostedZoneID, dnsName)
+		}
+	}
+	if recordValue != "" {
+		fmt.Printf("records for hostedZoneID %s with dnsName %s exists", hostedZoneID, dnsName)
+	}
+	return nil
+}
+
+func (c *Client) GetDNSRecord(dnsName string, hostedZoneID string) (string, error) {
+	params := &route53.ListResourceRecordSetsInput{
+		HostedZoneId:    aws.String(hostedZoneID),
+		MaxItems:        aws.String("1"),
+		StartRecordName: aws.String(dnsName),
+	}
+
+	resp, err := c.Route53Client.ListResourceRecordSets(params)
+	if err != nil {
+		return "", err
+	}
+	if len(resp.ResourceRecordSets) == 0 {
+		return "", errors.New(fmt.Sprintf("no record set exists for hostedZoneID %v with dnsName %v", hostedZoneID, dnsName))
```

Comment:
How about using fmt.Errorf here or errors.Errorf instead of using errors.New(fmt.Sprintf)?